perfkitbenchmarker/providers/aws/aws_emr.py

```diff
@@ -112,8 +132,40 @@ def _WaitUntilReady(self):
       time.sleep(READY_CHECK_SLEEP)
     return False
 
-  def SubmitJob(self, jarfile, classname, job_poll_interval=JOB_WAIT_SLEEP):
+  def _GetLogBase(self):
+    """Gets the base uri for the logs."""
+    cmd = self.cmd_prefix + ['describe-cluster', '--cluster-id',
+                             self.cluster_id]
+    stdout, _, _ = vm_util.IssueCommand(cmd)
+    result = json.loads(stdout)
+    if 'LogUri' in result['Cluster']:
+      self.logging_enabled = True
+      log_uri = result['Cluster']['LogUri']
+      if log_uri.startswith('s3n'):
+        log_uri = 's3' + log_uri[3:]
+      return log_uri
+    else:
+      return None
+
+  def _WaitForFile(self, filename, poll_interval=10):
+    """Wait for file to appear on s3."""
+    cmd = ['aws', 's3', 'ls', filename]
+
+    tries = 0
+    while tries < 30:
+      _, _, rc = vm_util.IssueCommand(cmd)
+      if rc == 0:
+        return True
+      else:
+        time.sleep(poll_interval)
+        tries += 1
+    return False
```

Comment:
It doesn't look like you check whether this method returns True or False (in submit job). If you are fine raising an exception instead of returning True/False, then you could use the vm_util.Retry decorator instead of having your own polling logic